hq/app/schedule/unrecognised/IamUnrecognisedUserJob.scala

```diff
@@ -33,9 +38,17 @@ class IamUnrecognisedUserJob(cacheService: CacheService, snsClient: AmazonSNSAsy
       logger.info(s"Running scheduled job: $description")
     }
 
-    val dummyJanusData: JanusData = ???
-
-    val janusUsers: Seq[String] = getJanusUsernames(dummyJanusData)
+    val janusUsers: Seq[String] = try {
+      val file = scala.io.Source
+        .fromInputStream(s3Client.getObject(s3BucketName.getOrElse(""), s3FileKey.getOrElse("")).getObjectContent)
```

Comment:
What does `getOrElse("")` really mean for us here? Do we want to try and load data from different locations in S3 if the configuration is missing?

I'd be tempted to structure this as a bunch of dependent steps, where fetching the data depends on being able to access both of these pieces of configuration. `flatMap` (and therefor a for-comprehension in Scala) is normally a good way to go when you want to express this kind of dependency chain, although it does mean choosing a consistent representation for failure.

Security HQ uses [`Attempt`](https://github.com/guardian/security-hq/blob/main/hq/app/utils/attempt/Attempt.scala) to handle failures elsewhere, so that's probably a good choice for the "effect" to use here, which means each of these steps will return an `Attempt` that represents either the successful value, or a description of the failure.

This is only looking at the diff though, when I look at the context around here I see that this is only one part of the work that's done, so a bigger for-comp that includes the work to get the credentials and perform side effects on the results is probably even better!

I spotted there's a nested foreach a bit lower down, which is always a bit of a red flag to my mind. It sounds like you're already looking at making this more testable, so that'll naturally mean those things get refactored away 👍